lib/logstash/filters/geoip.rb

```diff
@@ -82,139 +86,121 @@ class LogStash::Filters::GeoIP < LogStash::Filters::Base
   # of the same geoip_type share the same cache. The last declared cache size will 'win'. The reason for this is that there would be no benefit
   # to having multiple caches for different instances at different points in the pipeline, that would just increase the
   # number of cache misses and waste memory.
-  config :lru_cache_size, :validate => :number, :default => 1000
+  config :cache_size, :validate => :number, :default => 1000
 
   public
   def register
-    require "geoip"
-
     if @database.nil?
-      @database = ::Dir.glob(::File.join(::File.expand_path("../../../vendor/", ::File.dirname(__FILE__)),"GeoLiteCity*.dat")).first
-      if !File.exists?(@database)
-        raise "You must specify 'database => ...' in your geoip filter (I looked for '#{@database}'"
+      @database = ::Dir.glob(::File.join(::File.expand_path("../../../vendor/", ::File.dirname(__FILE__)),"GeoLite2-City.mmdb")).first
+
+      if @database.nil? || !File.exists?(@database)
+        raise "You must specify 'database => ...' in your geoip filter (I looked for '#{@database}')"
       end
     end
-    @logger.info("Using geoip database", :path => @database)
-    # For the purpose of initializing this filter, geoip is initialized here but
-    # not set as a global. The geoip module imposes a mutex, so the filter needs
-    # to re-initialize this later in the filter() thread, and save that access
-    # as a thread-local variable.
-    geoip_initialize = ::GeoIP.new(@database)
-
-    @geoip_type = case geoip_initialize.database_type
-    when GeoIP::GEOIP_CITY_EDITION_REV0, GeoIP::GEOIP_CITY_EDITION_REV1
-      :city
-    when GeoIP::GEOIP_COUNTRY_EDITION
-      :country
-    when GeoIP::GEOIP_ASNUM_EDITION
-      :asn
-    when GeoIP::GEOIP_ISP_EDITION, GeoIP::GEOIP_ORG_EDITION
-      :isp
-    else
-      raise RuntimeException.new "This GeoIP database is not currently supported"
-    end
 
-    @threadkey = "geoip-#{self.object_id}"
+    @logger.info("Using geoip database", :path => @database)
 
-    # This is wrapped in a mutex to make sure the initialization behavior of LOOKUP_CACHES (see def above) doesn't create a dupe
-    LOOKUP_CACHE_INIT_MUTEX.synchronize do
-      self.lookup_cache = LOOKUP_CACHES[@geoip_type] ||= LruRedux::ThreadSafeCache.new(1000)
+    db_file = JavaIO::File.new(@database)
+    begin
+      @parser = DatabaseReader::Builder.new(db_file).withCache(CHMCache.new(@cache_size)).build();
+    rescue Java::ComMaxmindDb::InvalidDatabaseException => e
+      # do nothing
```

Comment:
after trying this out, I notice that I receive the same error. This is received because the plugin continues in trying to filter, even thought it knows that the database file is invalid. I think the plugin should fail early and raise an exception, instead of just logging it.

here is what I am thinking:

The change in `register` to log and raise an exception
```ruby
@@ -124,7 +124,8 @@ class LogStash::Filters::GeoIP < LogStash::Filters::Base
     begin
       @parser = DatabaseReader::Builder.new(db_file).withCache(CHMCache.new(@cache_size)).build();
     rescue Java::ComMaxmindDb::InvalidDatabaseException => e
-      # do nothing
+      @logger.error("The Geoip2 MMDB database provided is invalid or corrupt.", :exception => e, :field => @source)
+      raise e
     end
   end # def register
```

The update to the spec to expect this failure to occur
```ruby
@@ -210,9 +210,9 @@ describe LogStash::Filters::GeoIP do
     context "should return the correct sourcefield in the logging message" do
       sample("ip" => "8.8.8.8") do
         expect(LogStash::Filters::GeoIP.logger).to receive(:error).with(anything, include(:field => "ip"))
-        subject
+        expect { subject }.to raise_error
       end
     end
   end
```